driver/ixsystems/iscsi.py

```diff
@@ -270,6 +273,18 @@ def create_volume_from_snapshot(self, volume, snapshot):
         self.common._create_iscsitarget(freenas_volume['target'],
                                         freenas_volume['name'])
 
+        # Promote image cache volume created by cinder service account
+        # by checking project_id is cinder service project and display name match
+        # image-[a-zA-Z0-9]+-[a-z0-9]+-[a-z0-9]+-[a-z0-9]+-[a-z0-9]+ pattern
+        # This is required because image cache volume cloned from the snapshot of first volume
+        # provisioned by this image from upstream cinder flow code
+        # Without promoting image cache volume, the first volume created can no longer be deleted
+        if (self.configuration.safe_get('image_volume_cache_enabled')
+            and self.common._is_service_project(volume['project_id'])
+            and re.match(r"image-[a-zA-Z0-9]+-[a-z0-9]+-[a-z0-9]+-[a-z0-9]+-[a-z0-9]+",
+                         volume['display_name'])):
+            self.common._promote_volume(freenas_volume['name'])
+
```

Comment:
Ensure only promote image cache volume created by service account (project_id is service project_id). This help to solve #31 issue.